wire-ios/Wire-iOS Tests/ConversationOptionsViewControllerTests.swift

```diff
@@ -360,8 +378,49 @@ final class ConversationOptionsViewControllerTests: BaseSnapshotTestCase {
         // for ConversationOptionsViewModel's delegate
         _ = ConversationGuestOptionsViewController(viewModel: viewModel)
         // Show the alert
-        let sut = viewModel.setAllowGuests(false)!
+        guard let sut = viewModel.setAllowGuests(false) else {
+            return XCTFail("This sut shouldn't be nil")
+        }
         // THEN
         try verify(matching: sut)
     }
+
+    // MARK: - Unit Tests
+
+    func testThatGuestLinkWithOptionalPasswordAlertShowIfApiVersionIsFourAndAbove() {
+        // GIVEN
+        BackendInfo.apiVersion = .v4
+        let config = MockOptionsViewModelConfiguration(allowGuests: true)
+        let viewModel = ConversationGuestOptionsViewModel(configuration: config)
+        let mock = MockConversationGuestOptionsViewModelDelegate()
+        mock.viewModelSourceViewPresentGuestLinkTypeSelection_MockMethod = { _, _, _ in }
+        mock.viewModelDidUpdateState_MockMethod = { _, _ in }
+        mock.viewModelDidReceiveError_MockMethod = { _, _ in }
+        viewModel.delegate = mock
+
+        // WHEN
+        viewModel.startGuestLinkCreationFlow()
+
+        // THEN
+        XCTAssert(mock.viewModelSourceViewPresentGuestLinkTypeSelection_Invocations.count == 1)
```

Comment:
Personally, `XCTAssertEqual` is more clear, but feel free to keep `XCTAssert`.

Reply:
Done!